.github/workflows/build.yml

```diff
@@ -2,134 +2,78 @@ name: Build
-             "${{ steps.variables.outputs.imagemagick_version }}"
-      - name: Test
-        run: "rpm -qp --requires rpms/ImageMagick-libs-${{ steps.variables.outputs.imagemagick_version }}.aarch64.rpm | grep -qEv 'libcdt|libcgraph|libgvc|libgs|libMagickCore|libMagickWand'"
-      - uses: actions/upload-artifact@v3
-        with:
-          name: rockylinux-arm64-rpms
-          path: rpms/*.rpm
-      - name: Deploy to nexus
-        uses: Alfresco/alfresco-build-tools/.github/actions/maven-deploy-file@7b0536b4e403e95365d83695c9ab3119b885ce36
-        if: startsWith(github.ref, 'refs/tags/v')
-        with:
-          group-id: org.imagemagick
-          artifact-id: imagemagick-distribution
-          repository-url: https://nexus.alfresco.com/nexus/content/repositories/thirdparty/
-          version: ${{ steps.build.outputs.built-version }}-ci-${{ steps.variables.outputs.imagemagick_release }}
-          generate-pom: false
-          file: rpms/ImageMagick-${{ steps.build.outputs.built-version }}.aarch64.rpm
-          classifier: el8-arm64
-          files: rpms/ImageMagick-libs-${{ steps.build.outputs.built-version }}.aarch64.rpm
-          classifiers: el8-libs-arm64
-          types: rpm
-          maven-username: ${{ secrets.NEXUS_USERNAME }}
-          maven-password: ${{ secrets.NEXUS_PASSWORD }}
+        run: docker run --name build buildenv ${{ steps.variables.outputs.imagemagick_version }} ${{ matrix.target_arch }}
 
-  build_centoslinux:
-    name: Centos 7
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Set build variables
-        id: variables
+      - name: Retrieve built packages
+        env:
+          IMAGEMAGICK_VERSION: ${{ steps.variables.outputs.imagemagick_version }}
         run: |
-          echo "imagemagick_version=$(cat ./.github/actions/imagemagick-version)" >> $GITHUB_OUTPUT
-          echo "imagemagick_release=$(cat ./.github/actions/release-version)" >> $GITHUB_OUTPUT
-      - name: Build
-        id: build
-        uses: ./.github/actions/centos-build
-        with:
-          args: 
-             "${{ steps.variables.outputs.imagemagick_version }}"
+          docker cp build:/root/rpmbuild/RPMS/ .
+          ls -l RPMS/
+
       - name: Test
-        run:  "rpm -qp --requires rpms/ImageMagick-libs-${{ steps.variables.outputs.imagemagick_version }}.x86_64.rpm | grep -qEv 'libcdt|libcgraph|libgvc|libgs|libMagickCore|libMagickWand'"
+        working-directory: RPMS/${{ matrix.target_arch }}
+        run: >-
+          rpm -qp --requires ImageMagick-libs-${{ steps.variables.outputs.imagemagick_version }}.${{ matrix.target_arch }}.rpm |
+          grep -qEv 'libcdt|libcgraph|libgvc|libgs|libMagickCore|libMagickWand'
+
       - uses: actions/upload-artifact@v3
         with:
-          name: centos-rpms
-          path: rpms/*.rpm
+          name: ${{ matrix.nexus_classifier }}-rpm
+          path: RPMS/${{ matrix.target_arch }}/*.rpm
```

Comment:
Any reason why we're uploading these artifacts? I don't see them being downloaded anywhere

Reply:
how do you test manually those packages before releasing them on nexus?